src/main/java/uk/sky/cirrus/CqlMigratorImpl.java

```diff
@@ -0,0 +1,125 @@
+package uk.sky.cirrus;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.ConsistencyLevel;
+import com.datastax.driver.core.QueryOptions;
+import com.datastax.driver.core.Session;
+import com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.sky.cirrus.locking.CassandraLockConfig;
+import uk.sky.cirrus.locking.CassandraLockingMechanism;
+import uk.sky.cirrus.locking.Lock;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+/**
+ * Standard implementation for {@code CqlMigrator}
+ *
+ * Implements Locking by providing mutex locking via the locks table.
+ * Client tries to acquire the lock and would only succeed for one thread.
+ * The inserted record will be deleted once the client releases the lock it holds, hence making way for a subsequent client thread to acquire the lock.
+ * Locking is done per client per keyspace. A client can obtain a lock on a different keyspace if no other client is holding a lock for that keyspace.
+ *
+ */
+
+public final class CqlMigratorImpl implements CqlMigrator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CqlMigratorImpl.class);
+
+    private final CassandraLockConfig lockConfig;
+
+    public CqlMigratorImpl(CassandraLockConfig lockConfig) {
+        this.lockConfig = lockConfig;
+    }
+
+    /**
+     * Allows cql migrate to be run on the command line.
+     * Set hosts, keyspace and directories using system properties.
+     */
+    public static void main(String[] args) {
+
+        final String hosts = System.getProperty("hosts");
+        final String keyspace = System.getProperty("keyspace");
+        final String directoriesProperty = System.getProperty("directories");
+        final String port = System.getProperty("port");
+
+        Preconditions.checkNotNull(hosts, "'hosts' property should be provided having value of a comma separated list of cassandra hosts");
+        Preconditions.checkNotNull(keyspace, "'keyspace' property should be provided having value of the cassandra keyspace");
+        Preconditions.checkNotNull(directoriesProperty, "'directories' property should be provided having value of the comma separated list of paths to cql files");
```

Comment:
I think we should enforce some sort of line length limit... long lines work fine for widescreen imacs but not for general public consumption. I suggest we add an .editorconfig at root level for it, 120 is good for me.